.github/workflows/deployment.yaml

```diff
@@ -33,3 +33,31 @@ jobs:
           # You can swap them out with your own user credentials.
           user_name: github-actions[bot]
           user_email: 41898282+github-actions[bot]@users.noreply.github.com
+
+      # erase everything below this comment if you do not wish to generate a pdf for the OER
+      # If you want to generate a pdf, alter lines 55 and 58.
+
+      - uses: actions/checkout@v4
+        with:
+          ref: gh-pages
+
+      - name: Install Prince
+        run: |
+          curl https://www.princexml.com/download/prince-14.2-linux-generic-x86_64.tar.gz -O
+          tar zxf prince-14.2-linux-generic-x86_64.tar.gz
+          cd prince-14.2-linux-generic-x86_64
+          yes "" | sudo ./install.sh
+
+      - name: Build PDF
+        run: npx docusaurus-prince-pdf -u https://seneca-ictoer.github.io/OERTemplate/
```

Comment:
One thing you could do here is use `env` variables to make the PDF generation conditional, and also set the name:

```yaml
name: deployment

on:
  push:
    branches: [main]

env:
  CREATE_PDF: true  # Set this to false if you don't want to create a PDF
  PDF_NAME: OERTemplate.pdf  # Set the name of the PDF file here

jobs:
  gh-release:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - uses: actions/setup-node@v3
        with:
          node-version: '18.x'
          cache: 'yarn'
      - name: Install dependencies
        run: yarn install --frozen-lockfile
      - name: Build website
        run: yarn build

      - name: Deploy to GitHub Pages
        uses: peaceiris/actions-gh-pages@v3
        with:
          github_token: ${{ secrets.OER_DEPLOY }}
          publish_dir: ./build
          user_name: github-actions[bot]
          user_email: 41898282+github-actions[bot]@users.noreply.github.com

      - name: Generate PDF
        if: env.CREATE_PDF == 'true'  # Only run this step if CREATE_PDF is true
        run: |
          # Checkout gh-pages branch
          git checkout gh-pages

          # Install Prince
          curl https://www.princexml.com/download/prince-14.2-linux-generic-x86_64.tar.gz -O
          tar zxf prince-14.2-linux-generic-x86_64.tar.gz
          cd prince-14.2-linux-generic-x86_64
          yes "" | sudo ./install.sh

          # Build PDF
          npx docusaurus-prince-pdf -u https://seneca-ictoer.github.io/OERTemplate/

          # Rename PDF
          mv ./pdf/seneca-ictoer.github.io-OERTemplate.pdf ./pdf/${{ env.PDF_NAME }}

          # Push PDF
          git config user.name github-actions
          git config user.email github-actions@github.com
          git add ./pdf
          git commit -m "add pdf file"
          git push origin gh-pages
```

NOTE: I haven't run this, but that's the idea...

Reply:
Let me try that out... but definitely more elegant than removing all the lines...

Reply:
Probably we can move the URL into an env var too to make this more portable.